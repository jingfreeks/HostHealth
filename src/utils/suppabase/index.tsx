export {default as supabase} from './suppabase';
