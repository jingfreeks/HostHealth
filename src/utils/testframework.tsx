import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
// import {QueryClient, QueryClientProvider} from '@tanstack/react-query';
// // import {DARK_THEME} from '@/utils/constants';
// // import {ThemeProvider} from 'styled-components/native';

type MockProviderType = {
  children?: React.ReactNode;
  store?: object;
};
export const MockProvider = (props: MockProviderType) => {
  const {store, children} = props;
  const mockStore = configureStore([thunk]);
  const mocked = mockStore(store);
  return <Provider store={mocked}>{children}</Provider>;
};

export const testingProps = (reference: string): unknown => ({
  testID: reference,
  accessibilityLabel: reference,
});

// const queryClient = new QueryClient({
//   defaultOptions: {
//     queries: {
//       retry: false,
//     },
//   },
// });
// export const wrapper = ({children}) => (
//   <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
// );
