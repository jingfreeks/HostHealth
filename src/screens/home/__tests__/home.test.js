import React from 'react';
import {waitFor,act} from '@testing-library/react-native'
import mockAsyncStorage from '@react-native-async-storage/async-storage/jest/async-storage-mock';
import {renderWithProviders} from '@/utils/testframeworknew';
import Home from '../home';

describe('Home Screen', () => {
  it('Should work as expected to get snapshot', () => {
    const all = renderWithProviders(<Home />);
    waitFor(() => {
      expect(all.toJSON()).toMatchSnapshot();
    });
    const el=all.getByTestId('HomeScreenOnRefreshTestId')
  });

  it('Should work as to trigger refresh function', async() => {
    const all = renderWithProviders(<Home />);

    const el=all.getByTestId('HomeScreenOnRefreshTestId')
    expect(el).toBeDefined();
    const {refreshControl} = el.props;
    await act(async () => {
      refreshControl.props.onRefresh();
    });
    // waitFor(() => {
    //   expect(all.toJSON()).toBeTruthy();
    // });
  });
});

