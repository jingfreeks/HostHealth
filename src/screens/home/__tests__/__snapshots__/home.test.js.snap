// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen Should work as expected to get snapshot 1`] = `
<RCTScrollView
  accessibilityLabel="HomeScreenOnRefreshTestId"
  refreshControl={
    <RefreshControlMock
      onRefresh={[Function]}
      refreshing={false}
    />
  }
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
      },
    ]
  }
  testID="HomeScreenOnRefreshTestId"
>
  <RCTRefreshControl />
  <View>
    <View
      style={
        [
          {
            "alignself": "stretch",
            "borderColor": "#99E1E1",
            "borderRadius": 10,
            "borderWidth": 2,
            "flexDirection": "row",
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginBottom": 10,
              "marginLeft": 10,
              "marginTop": 10,
            },
          ]
        }
      >
        <Image
          resizeMode="stretch"
          size={50}
          source={
            {
              "testUri": "../../../src/assets/htimecard.png",
            }
          }
          style={
            [
              {
                "height": 117.70588235294117,
                "width": 98.08823529411764,
              },
            ]
          }
        />
      </View>
      <View
        style={
          [
            {
              "alignSelf": "center",
              "flexBasis": 0,
              "flexGrow": 3,
              "flexShrink": 1,
              "justifyContent": "center",
              "marginBottom": 10,
              "marginTop": 10,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#0e3f6c",
                "fontFamily": "Avenir",
                "fontSize": 39.23529411764706,
                "fontWeight": "900",
              },
            ]
          }
        >
          Timecards
        </Text>
        <Text
          style={
            [
              {
                "color": "#0e3f6c",
                "fontFamily": "Avenir",
                "fontSize": 12,
                "fontWeight": "400",
              },
            ]
          }
        >
          Easily upload images for 
your timecards for quick processing.
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "alignself": "stretch",
            "borderColor": "#99E1E1",
            "borderRadius": 10,
            "borderWidth": 2,
            "flexDirection": "row",
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginBottom": 10,
              "marginLeft": 10,
              "marginTop": 10,
            },
          ]
        }
      >
        <Image
          resizeMode="stretch"
          size={50}
          source={
            {
              "testUri": "../../../src/assets/certificate.png",
            }
          }
          style={
            [
              {
                "height": 117.70588235294117,
                "width": 98.08823529411764,
              },
            ]
          }
        />
      </View>
      <View
        style={
          [
            {
              "alignSelf": "center",
              "flexBasis": 0,
              "flexGrow": 3,
              "flexShrink": 1,
              "justifyContent": "center",
              "marginBottom": 10,
              "marginTop": 10,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#0e3f6c",
                "fontFamily": "Avenir",
                "fontSize": 39.23529411764706,
                "fontWeight": "900",
              },
            ]
          }
        >
          Credentialing
        </Text>
        <Text
          style={
            [
              {
                "color": "#0e3f6c",
                "fontFamily": "Avenir",
                "fontSize": 12,
                "fontWeight": "400",
              },
            ]
          }
        >
          Add information for backgournd
check and 19. (Links to Essium)
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "marginTop": 20,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#0e3f6c",
                "fontFamily": "Avenir",
                "fontSize": 35.311764705882354,
                "fontWeight": "500",
              },
            ]
          }
        >
          SUGGESTED
        </Text>
      </View>
      <View
        style={
          [
            {
              "alignItems": "center",
              "alignSelf": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <RCTSafeAreaView
          style={
            [
              {},
            ]
          }
        >
          <View
            alignType="center"
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "flexDirection": "column",
                },
              ]
            }
          >
            <Image
              resizeMode="contain"
              size={150}
              source={
                {
                  "testUri": "../../../src/assets/suggestempty.png",
                }
              }
              style={
                [
                  {
                    "height": 294.2647058823529,
                    "width": 294.2647058823529,
                  },
                ]
              }
            />
            <Text
              style={
                [
                  {
                    "color": "#000000",
                    "fontFamily": "Avenir",
                    "fontSize": 16,
                    "fontWeight": "400",
                  },
                ]
              }
            >
              No suggested jobs matches with your profile
            </Text>
          </View>
        </RCTSafeAreaView>
      </View>
      <View
        style={
          [
            {
              "marginTop": 20,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#0e3f6c",
                "fontFamily": "Avenir",
                "fontSize": 35.311764705882354,
                "fontWeight": "500",
              },
            ]
          }
        >
          POPULAR CITIES
        </Text>
        <View
          style={
            [
              {
                "alignItems": "center",
                "alignSelf": "center",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "justifyContent": "center",
              },
            ]
          }
        >
          <RCTSafeAreaView
            style={
              [
                {},
              ]
            }
          >
            <View
              alignType="center"
              style={
                [
                  {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "flexDirection": "column",
                  },
                ]
              }
            >
              <Image
                resizeMode="contain"
                size={150}
                source={
                  {
                    "testUri": "../../../src/assets/pcitiesempty.png",
                  }
                }
                style={
                  [
                    {
                      "height": 294.2647058823529,
                      "width": 294.2647058823529,
                    },
                  ]
                }
              />
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "Avenir",
                      "fontSize": 16,
                      "fontWeight": "400",
                    },
                  ]
                }
              >
                No cities matches your preferred locations
              </Text>
            </View>
          </RCTSafeAreaView>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
